Source/Core/Common/Hash.cpp

```diff
@@ -530,4 +532,19 @@ void SetHash64Function()
     ptrHashFunction = &GetMurmurHash3;
   }
 }
+
+std::string MakeGameID(std::string file_name)
+{
+  const char* file_name_c = file_name.c_str();
+  unsigned long hash = 5381;
+  int c = *file_name_c;
```

Comment:
```suggestion
  auto c = static_cast<u64>(file_name_c[i]);
```

So that there's no implicit sign conversion below.